samples/boards/96b_argonkey/microphone/README.rst

```diff
@@ -0,0 +1,158 @@
+in either one of the following two ways:
+
+- mezzanine mode, plugging the ArgonKey to HiKey board thru its 96Board
+  low-speed connector
+- standalone mode, supplying 5V directly on P1 connector
+
+References
+**********
+
+- :ref:`96b_argonkey`
+
+Building and Running
+********************
+
+.. zephyr-app-commands::
+    :zephyr-app: samples/boards/96b_argonkey/microphone
+    :host-os: unix
+    :board: 96b_argonkey
+    :goals: run
+    :compact:
+
+Sample Output
+=============
+
+The example acquires 5s of audio and prints out the PCM stream on COM port.
+A USB to TTL 1V8 serial cable may be attached to the low speed connector on
+the back of the board on P3.5 (TX) and P3.7 (RX).
+
+As soon as the acquisition starts the green LED glows. At the end of the
+acquisition both the green and red LEDs go on and the PCM stream is sent
+on COM port. When the output is completed the green LED goes off and the red
+LED stays on. The process can be reiterated by pressing the RST button.
+
+- audio acquisition starts:     GRN on  - RED off
+- audio acquisition ends:       GRN on  - RED on
+- audio output ends:            GRN off - RED on
+
+The characteristics of the PCM audio are hardcoded in the example:
+
+- 16KHz sample rate
+- 16 bits per sample
+- 1 channel (mono), as only 1 microphone is available
+
+Five seconds of acquisition at a 16KHz sampling rate yields 80,000 16-bit samples.
+The microphone PDM requested clock should lead the MP34DT05 driver to select an
+oversampling/decimation factor equal to 128, resulting in a 2.048MHz bit clock.
+
+See configuration in file :file:`samples/boards/96b_argonkey/microphone/src/main.c`:
+
+.. code-block:: c
+
+   #define AUDIO_FREQ          16000
+   #define CHAN_SIZE           16
+   #define PCM_BLK_SIZE_MS     ((AUDIO_FREQ/1000) * sizeof(s16_t))
+
+   #define NUM_MS              5000
+
+   K_MEM_SLAB_DEFINE(rx_mem_slab, PCM_BLK_SIZE_MS, NUM_MS, 1);
+
+   struct pcm_stream_cfg mic_streams = {
+        .pcm_rate = AUDIO_FREQ,
+        .pcm_width = CHAN_SIZE,
+        .block_size = PCM_BLK_SIZE_MS,
+        .mem_slab = &rx_mem_slab,
+   };
+
+   struct dmic_cfg cfg = {
+        .io = {
+            /* requesting a pdm freq around 2MHz */
+            .min_pdm_clk_freq = 1800000,
+            .max_pdm_clk_freq = 2500000,
+        },
+        .streams = &mic_streams,
+        .channel = {
+            .req_num_chan = 1,
+        },
+   };
+
+.. note:: It is possible to change the AUDIO_FREQ to 32000 acquiring only 2500 ms. In this
+   case the oversampling/decimation factor will be 64.
+
+At the end of the acquisition the PCM data will be printed on the terminal
+emulator in either binary or ASCII format. The output is controlled by
+following macro, off by default, in :file:`samples/boards/96b_argonkey/microphone/src/main.c`:
+
+* :c:macro:`PCM_OUTPUT_IN_ASCII`
+
+Binary PCM Output
+-----------------
+
+The ttyUSB0 port must be configured in raw mode to avoid having
+characters 'cooked' out.
```

Comment:
'cooked'? Is this a well-known term?

Reply:
Yes. Basically TTY drivers has two modes 'raw' vs 'cooked'. In this sample we need raw otherwise some characters may be processed by the driver itself and not sent out, corrupting the audio stream (you will hear a lot of noise).